runtime/zsr/find.hpp

```diff
@@ -51,24 +54,58 @@ DECL_ITER(ServiceMethod,   Service,     methods)
 
 #undef DECL_ITER
 
+template <class _Type, class _Root>
+const _Type* find(const _Root& r, const std::string_view& ident)
+{
+    auto [begin, end] = impl::child_iter<_Type>::get(r);
+    auto iter = std::find_if(begin, end, [&](const auto& c) {
+        return c.ident == ident;
+    });
+    if (iter != end)
+        return &(*iter);
+    return nullptr;
+}
+
+template <class _Root, class _Type>
+struct TypeHierarchyHelper
+{
+    template<class _ReverseIter>
+    static const _Type* find(_Root const& x, _ReverseIter begin, _ReverseIter end)
+    {
+        auto parent = TypeHierarchyHelper<_Root, typename child_iter<_Type>::ParentType>::find(x, begin+1, end);
+        if (!parent)
+            return nullptr;
+        return impl::find<_Type>(*parent, *begin);
+    }
+};
+
+template <class _Type>
+struct TypeHierarchyHelper<typename child_iter<_Type>::ParentType, _Type>
+{
+    template<class _Root, class _ReverseIter>
+    static auto find(_Root const& parent, _ReverseIter begin, _ReverseIter end)
+    {
+        auto ident = stx::join(end.base(), begin.base(), ".");
+        return impl::find<_Type>(parent, ident);
+    }
+};
+
 } // namespace impl
 
 /**
  * Utility functions for searching metadata.
  *
  * Example:
  *   auto field = zsr::find<zsr::Field>(compound, "name")
+ *
+ * Example (with recursive field resolution):
+ *   auto field = zsr::find<zsr::Field>(compound, "member.name")
```

Comment:
Parameter `compound` should be replaced with `package` or `zsr::packages()`, “member.name” with “compound.member”